### PR TITLE
removed 'recommended' from the datadir set up wording

### PR DESCRIPTION
**Pull request by ale-cristofori**
This PR applies a change in the Geoserver docs, for the part that documents how to externalize the data-directory.
The wording contained the word 'recommended' for the set up of the context-parameter in the web.xml file.
The word 'recommended' was removed since it could be mistaken by the readers, there is not a 'recommended'  method, but is explained as an alternative.
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by aaime** [comment]
Looks good to me, the configuration in web.xml is actually annoying when upgrading, as one has to patch the file again.

**Comment by jodygarnett** [comment]
why remove? setting up a geoserver.xml file to define data directory location is exactly what I recommend. 

The other techniques, system property and environmental variable, may accidentally be picked up by another web app.

File: doc/en/user/source/datadirectory/setting.rst [hunk 1]
```diff
@@ -47,7 +47,7 @@ For the Mac OS X install, set the ``GEOSERVER_DATA_DIR`` environment variable to
 Web archive
 -----------
 
-When running a GeoServer WAR inside a servlet container, the data directory can be specified in a number of ways. The recommended method is to set a **servlet context parameter**. An alternative is to set a **Java system property**.
+When running a GeoServer WAR inside a servlet container, the data directory can be specified in a number of ways. One alternative is to set a **servlet context parameter**, another alternative is to set a **Java system property**.
```
Comment (jodygarnett) [inline doc/en/user/source/datadirectory/setting.rst:50]:
```suggestion
When running a GeoServer WAR inside a servlet container, the data directory can be specified in a number of ways. 

See :ref:`application_properties` section for details on setting this location using java system property, web context parameter, or environmental variable.
```

**Comment by aaime** [comment]
Nobody said to "remove" it, just to remove its recommendation. The system variable is unlikely to be picked from another application, because it would have to be another GeoServer to make sense of it... two GeoServer in the same Tomcat seems to be pretty dumb (high memory usage, does not undeploy cleanly...). The environment variables are the default for docker images instead (normally Tomcat is literally the main process in the docker image).

Your alternative suggestion is good though, I've merged it.

**Comment by petersmythe** [comment]
@aaime this PR wasn't merged, so I have gone ahead and done that.

**Pull request merged**